plugins/com.aptana.js.core/src/com/aptana/js/internal/core/build/JSParserValidator.java

```diff
@@ -89,6 +87,9 @@ public void buildFile(final BuildContext context, IProgressMonitor monitor)
 			Collection<IParseError> parseErrors = context.getParseErrors();
 			if (!CollectionsUtil.isEmpty(parseErrors))
 			{
+				fLocation = context.getURI();
```

Comment:
I don't see how this would avoid an NPE just by moving this down here.

Maybe keep them where they were and just guard the fPath assignment to be null or empty string (not sure it making it null would cause another NPE down the line) if fLocation is null?